app/models/event.rb

```diff
@@ -0,0 +1,60 @@
+# Object representing relevant actions taken by a case  manager
+class Event
+  include Mongoid::Document
+  include Mongoid::Timestamps
+  extend Enumerize
+
+  EVENT_TYPES = [
+    'Reached patient',
+    "Couldn't reach patient",
+    'Left voicemail',
+    'Pledged',
+    'Unknown action'
+  ].freeze
+
+  # Fields
+  field :cm_name, type: String
+  enumerize :event_type,
+            in:      EVENT_TYPES.map(&:to_sym),
+            default: 'Unknown action'.to_sym
+  # See config/initializers/env_var_contants.rb
+  enumerize :line, in: LINES, default: LINES[0]
+  field :patient_name, type: String
+  field :patient_id, type: String
+  field :pledge_amount, type: Integer
+
+  # Validations
+  validates :event_type, inclusion: { in: EVENT_TYPES }
+  validates :cm_name, :patient_name, :patient_id, :line, presence: true
+  validates :pledge_amount, presence: true, if: :pledged_type?
+
+  def glyphicon
+    return 'thumbs-up' if event_type == 'Pledged'
+    'earphone'
+  end
+
+  def event_text
+    case event_type
+    when 'Pledged'
+      "sent a $#{pledge_amount} pledge for"
+    when 'Left voicemail'
+      'left a voicemail for'
+    when "Couldn't reach patient"
+      "called, but couldn't reach"
+    when 'Reached patient'
+      'reached'
+    end
+  end
+
+  # Clean events older than three weeks
+  def self.destroy_old_events
+    Event.where(:created_at.lte => 3.weeks.ago)
```

Comment:
Huh. This is the first time Rails' readability has backfired for me. I read this the first ~3 times as "events created less than three weeks ago", instead of the correct "Events where created_at is before three weeks ago".

Reply:
This also bit me while I was writing it! There was a good five minutes where I was positive I was losing my mind.